src/browser/components/MainPage.jsx

```diff
@@ -598,6 +602,39 @@ export default class MainPage extends React.Component {
       this.switchToTabForCertificateRequest(certificateRequests[0].server);
     }
   };
+
+  showDownloadCompleteNotification = async (event, item) => {
+    const title = 'Download Complete';
+
+    const notification = await this.dispatchNotification(title, item.title);
+    notification.onclick = () => {
+      shell.showItemInFolder(item.defaultPath.normalize());
+    };
+  }
+
+  dispatchNotification = async (title, body, channel, teamId, silent) => {
+    const permission = await Notification.requestPermission();
+    if (permission !== 'granted') {
+      log.error('Notifications not granted');
+      return null;
+    }
+    const notification = new Notification(title, {
+      body,
+      tag: body,
+      icon: appIconURL,
+      requireInteraction: false,
+      silent,
+    });
+    notification.onclick = () => {
+      this.webviewRef.current.send('notification-clicked', {channel, teamId});
```

Comment:
why send channel and teamId if it's undefined? shouldn't we use a different event? are we notifying the webapp for some reason?

Reply:
I think this is just leftover code from the other notification block.

Reply:
yes, I was thinking of extracting it to a util function, is there a preferred place to do that?

Reply:
You can add it to `src/util/util.js`